grayscale.py

```diff
@@ -1,764 +1,549 @@
+        #
+        # DISCLAIMER: The following comments relating to the hardware
+        # may not be technically correct, and may instead only be
+        # effectively correct.
+        #
+        # We keep the display synchronised by resetting the row counter
+        # before each frame and then outputting a frame of 57 rows.
+        # This is 17 rows past the 40 of the actual display.
+        #
+        # Prior to loading in the frame we park the row counter at row 0
+        # and wait for the nominal time for 8 rows to be output. This
+        # seems to provide enough time for the row counter to reach row 0
+        # before it sticks there. Note: the row counter may jump then stick.
+        #
+        # The 'parking' is done by setting the number of rows (aka
+        # 'multiplex ratio') to 1 row. This is an invalid setting
+        # according to the datasheet but seems to still have the desired
+        # effect.
         # 0xa8,0    Set multiplex ratio to 1
         # 0xd3,52   Set display offset to 52
-        self.pre_frame_cmds = bytearray([0xa8,0, 0xd3,52])
-        # Once the frame has been loaded into the display controller's GDRAM, we
-        # set the controller to output 57 rows, and then delay for the nominal
-        # time for 48 rows to be output.
-        # Considering the 17 row 'buffer space' after the real 40 rows, that puts
-        # us around halfway between the end of the display, and the row at which
-        # it would wrap around.
-        # By having 8.5 rows either side of the nominal timing, we can absorb any
-        # variation in the frequency of the display controller's RC oscillator as
-        # well as any timing offsets introduced by the Python code.
-        # 0xd3,x    Set display offset. Since rows are scanned in reverse, the
-        #           calculation must work backwards from the last controller row.
+        self._preFrameCmds = bytearray([0xa8,0, 0xd3,52])
+        # Once the frame has been loaded into the display controller's
+        # GDRAM, we set the controller to output 57 rows, and then delay
+        # for the nominal time for 48 rows to be output.
+        # Considering the 17 row 'buffer space' after the real 40 rows,
+        # that puts us around halfway between the end of the display, and
+        # the row at which it would wrap around.
+        # By having 8.5 rows either side of the nominal timing, we can
+        # absorb any variation in the frequency of the display controller's
+        # RC oscillator as well as any timing offsets introduced by the
+        # Python code.
+        # 0xd3,x    Set display offset. Since rows are scanned in reverse,
+        #           the calculation must work backwards from the last
+        #.          controller row.
         # 0xa8,57-1 Set multiplex ratio to 57
-        self.post_frame_cmds = bytearray([0xd3,40+(64-57), 0xa8,57-1])
+        self._postFrameCmds = bytearray([0xd3,40+(64-57), 0xa8,57-1])
```

Comment:
I wanted to move towards consistency. It was already pretty mixed up on the main branch, so it seemed it could be argued to go either way. Given that it was already a bit mixed up, I chose to match the much more prominent style within the Thumby API like here: https://github.com/TinyCircuits/TinyCircuits-Thumby-Code-Editor/blob/master/ThumbyGames/lib/thumbyGraphics.py

They already have a little mixing but it very much leans towards mixedCase. Their inclusion of the ssd1306 under the display sub-property threw a spanner in the works for consistency, which ended up polluting my results little where I was trying  to match compatibility. I thought briefly about putting the ssd1306 compatibility in a subclass under .display to hide it only where the compatibility was needed but when I ran that scenario, it wasn't breathtakingly pretty.

I'm happy to go in any direction. I'll change it all to whatever you and Timendus want. But I'd love to know what we are matching, what our coding style preference is. I love PEP8, but, even it likes consistency more, and if I was making the decision on my own, I would entirely match the ThumbyAPI, as best I could.

But that aside, let me know what the code style is that we are targeting and I will adjust.